### PR TITLE
[Storage] Support Managepolicy policy on 2018-03-01-preview

### PR DESCRIPTION
**Pull request by blueww**
## Description

This PR is generated from swagger in private repo, will update to public repo when the swagger PR for public repo is merged.
But the code should be the same, so please help to start review now. 
The feature target to release on 5/30.

The swagger PR:
Private: pass review on private repo: Azure/azure-rest-api-specs-pr#465
Public: just sent to public repo: https://github.com/Azure/azure-rest-api-specs/pull/3137 

Rest API Spec: https://microsoft-my.sharepoint.com/:w:/r/personal/yzheng_ntdev_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B3E66A1C5-12C8-4ACB-BCF0-D59B015DE394%7D&file=DLMPreviewDoc.docx&action=default&mobileredirect=true

Breaking:
Per onesdk team requirement in issue Azure/azure-rest-api-specs#3024, change Usage to Usages, since operation ID should not be singleton.

Upgrade SDK version to 8.0.0-preview since the breaking.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by blueww** [comment]
@abhijeetgaiha 

The build failed since a Test case fail.
It seems the failure is not related with my code change, would you please help to fix it and build?

> 21:05:12   [xUnit.net 00:00:01.8712997]     TestFramework.Tests.TestEnvironment.TestEnvironmentTests.DefaultTenantInTestEnvironment [FAIL]
> 21:05:12   [xUnit.net 00:00:01.8735540]       System.AggregateException : One or more errors occurred. (AADSTS70002: Error validating credentials. AADSTS50012: Invalid client secret is provided.
> 21:05:12   [xUnit.net 00:00:01.8738628]       Trace ID: d8de02bd-3b48-4661-be8a-a0d67c6a1d00
> 21:05:12   [xUnit.net 00:00:01.8739711]       Correlation ID: 671433c7-97df-44d2-865c-ed0731794485
> 21:05:12   [xUnit.net 00:00:01.8740656]       Timestamp: 2018-05-25 04:05:11Z)

File: src/SDKs/Storage/Management.Storage/Microsoft.Azure.Management.Storage.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
 		<PackageId>Microsoft.Azure.Management.Storage</PackageId>
 		<Description>Microsoft Azure Management Storage Library</Description>
 		<AssemblyName>Microsoft.Azure.Management.Storage</AssemblyName>
-		<Version>7.2.0-preview</Version>
+		<Version>8.0.0-preview</Version>
 		<PackageTags>Microsoft Azure Storage management;Storage;Storage management;</PackageTags>
 		<PackageReleaseNotes>See https://aka.ms/asdotnetsdkchangelog for release notes.</PackageReleaseNotes>
```
Comment (dsgouda) [inline src/SDKs/Storage/Management.Storage/Microsoft.Azure.Management.Storage.csproj]: Please confirm that the package release notes are updated

File: src/SDKs/Storage/Storage.Tests/Tests/StorageAccountTests.cs [hunk 2]
```diff
@@ -1170,12 +1148,27 @@ public void StorageAccountUpdateEncryptionTest()
                 Assert.Equal(true, account.Encryption.Services.File.Enabled);
                 Assert.NotNull(account.Encryption.Services.File.LastEnabledTime);
 
-                // 2. Update storage encryption by enable Blob Encyrption
+                //// 2. Explicitly disable file encryption service.
```
Comment (dsgouda) [inline src/SDKs/Storage/Storage.Tests/Tests/StorageAccountTests.cs]: We generally discourage commented code. It's best to move this code to a different test and mark it as `Skipped`

File: src/SDKs/Storage/Management.Storage/Microsoft.Azure.Management.Storage.csproj (hunk 1 above)
Comment (dsgouda) [reply src/SDKs/Storage/Management.Storage/Microsoft.Azure.Management.Storage.csproj]: nit: Is there a reason you plan to release a major version, you can release preview breaking changes by bumping the minor version too

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/_metadata/storage_resource-manager.txt [hunk 3]
```diff
@@ -1,9 +1,9 @@
-2018-05-07 05:06:06 UTC
+2018-05-25 03:21:52 UTC
 
 1) azure-rest-api-specs repository information
-GitHub user: Azure
-Branch:      master
-Commit:      96d9da589ae6610fc816fca6b25288519974c83a
+GitHub user: blueww
```
[the remark this answers is not in the document]
Comment (blueww) [reply src/SDKs/_metadata/storage_resource-manager.txt]: 1. As I mentioned in the PR, I will regerate the SDK code from the public swagger repo, when the swagger PR is merged. But first send it for review.

2. Ｉ have already added it before. Not sure if the current one is good for you.

File: src/SDKs/Storage/Storage.Tests/Tests/StorageAccountTests.cs (hunk 2 above)
Comment (blueww) [reply src/SDKs/Storage/Storage.Tests/Tests/StorageAccountTests.cs]: Good catch.
Fixed.

File: src/SDKs/Storage/Management.Storage/Microsoft.Azure.Management.Storage.csproj (hunk 1 above)
Comment (blueww) [reply src/SDKs/Storage/Management.Storage/Microsoft.Azure.Management.Storage.csproj]: 1. Where's the package release notes? Do you mean the change log?  I have updated it.
2. I update the major version just for the breaking change. Although we still in preview, we use the GA quality bar.

**Comment by dsgouda** [comment]
@blueww Please pull down latest upstream changes and update the PR

**Review by dsgouda — APPROVED** [review]
LGTM, will merge on CIs passing

**Pull request merged**